### PR TITLE
jmap_mailbox: prefer to report changes in updated if modseqs match

### PR DESCRIPTION
**Pull request by rsto**
Mailbox renames result in two mboxlist entries with the same uniqueid and modseq, one of them being deleted. Until now, Mailbox/changes either reports this mailbox id in `updated` or `destroyed`, depending on mboxlist traversal order. This patch changes this to always prefer reporting equal modseq changes in `updated` over `created` or `destroyed`.

Tested in https://github.com/cyrusimap/cassandane/commit/08d5b4227964c43b5986a7e6e2dbfe4c2d34acf1

### CODE REVIEW

File: imap/jmap_mailbox.c [hunk 1]
```diff
@@ -4158,6 +4158,20 @@ static int _mbox_changes_cb(const mbentry_t *mbentry, void *rock)
         *(data->only_counts_changed) = 0;
     }
 
+    /* Determine where to report that update. Note that we even report
+     * hidden mailboxes in order to allow clients remove unshared ando
```
Comment (ksmurchison) [inline imap/jmap_mailbox.c:4162]: Small typo "ando"

File: imap/jmap_mailbox.c [hunk 2]
```diff
@@ -4158,6 +4158,20 @@ static int _mbox_changes_cb(const mbentry_t *mbentry, void *rock)
...
+     * deleted mailboxes */
+    json_t *dest = NULL;
+    if ((mbentry->mbtype & MBTYPE_DELETED) || !jmap_hasrights_mbentry(req, mbentry, JACL_LOOKUP)) {
+        if (mbentry->createdmodseq <= data->since_modseq)
+            dest = data->destroyed;
+    } else {
+        if (mbentry->createdmodseq <= data->since_modseq)
```
Comment (ksmurchison) [inline imap/jmap_mailbox.c:4169]: Nit: Can't this if/else be moved up to be else if/else ?

**Review by ksmurchison — APPROVED** [review]
Other than 2 nits, looks sane to me

File: imap/jmap_mailbox.c (hunk 1 above)
Comment (rsto) [reply imap/jmap_mailbox.c]: fixed

File: imap/jmap_mailbox.c (hunk 2 above)
Comment (rsto) [reply imap/jmap_mailbox.c]: thanks. I simplified that if expression.

**Review by ksmurchison — APPROVED** [review]

**Review by brong — COMMENTED** [review]
This looks good - though hopefully eventually this will all be looking at the "I" keys in the new style mailboxes.db, and hence we can tell if the mailbox still exists or not from there!

**Pull request merged**